### PR TITLE
Content management: errata-related fixes

### PR DESCRIPTION
**Pull request by hustodemon**
## What does this PR change?
 
This PR contains a couple of fixes related to errata handling in content management. Commit messages are mostly self-explanatory.

Explicit note to the last commit (Fix: Truncating errata - take cloned errata into account):
This was a bug in the content management errata truncating: when building a project, errata are cloned from the source channel to target channel in the first environment. If for some reason, an erratum is removed from the source channel and the project is built again, its clone of this erratum should also be removed from the target channel.
 
 ## GUI diff
 
 No difference.
 
 Before:
 
 After:
 
 - [x] **DONE**
 
 ## Documentation
 - No documentation needed: bugfixes
 
 - [x] **DONE**
 
 ## Test coverage
 - Unit tests were added
 
 - [x] **DONE**
 
 
 ## Changelogs
 
 If you don't need a changelog check, please mark this checkbox:
 
 - [x] No changelog needed
 
 If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)
 
 
 ## Re-run a test
 
 If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:
 
 - [ ] Re-run test "changelog_test"
 - [ ] Re-run test "backend_unittests_pgsql"
 - [ ] Re-run test "java_lint_checkstyle"		 
 - [ ] Re-run test "java_pgsql_tests"		 
 - [ ] Re-run test "ruby_rubocop"
 - [ ] Re-run test "schema_migration_test_oracle"
 - [ ] Re-run test "schema_migration_test_pgsql"		 
 - [ ] Re-run test "susemanager_unittests"
 - [ ] Re-run test "javascript_lint"

### CODE REVIEW

File: java/code/src/com/redhat/rhn/manager/errata/ErrataManager.java [hunk 1]
```diff
@@ -320,11 +322,18 @@ public static void truncateErrata(Channel srcChannel, Channel tgtChannel, User u
         Set<Errata> srcErrata = new HashSet<>(ErrataFactory.listByChannel(user.getOrg(), srcChannel));
         Set<Errata> tgtErrata = new HashSet<>(ErrataFactory.listByChannel(user.getOrg(), tgtChannel));
 
-        // select errata that are exclusively in tgt channel
-        tgtErrata.removeAll(srcErrata);
+        // let's remove errata that aren't in the source errata nor is their original
+        tgtErrata.stream()
+                .filter(e -> !(srcErrata.contains(e) ||
+                        asCloned(e).map(er -> srcErrata.contains(er.getOriginal())).orElse(false)))
```
[the remark this answers is not in the document]
Comment (hustodemon) [reply java/code/src/com/redhat/rhn/manager/errata/ErrataManager.java]: The situation like you described will not happen in the Content Management scenario: when we are cloning the errata using the `CloneErrataAction`, it'll look like this:

* `channelA` (source channel) contains `errataA`
* `channelB` (channel in the 1st environment after build) contains `cloneErrataA`
* `channelC` (channel in the 2nd environment after promote) contains `cloneErrataA`

Now, hypothetically, when the situation described by you happened (let's say that `truncateErrata` is used outside content management in the future), I think both cases are valid:

Input conditions:
* The channel/errata configuration as you described
* `cloneErrataA` is removed from the `channelB`

Then we want to truncate `channelC` to `channelB`:
* Scenario 1: `cloneOfCloneErrataA` original points to `errataA` (we say that the `cloneOfCloneErrataA` and `cloneOfERrataA` are "brothers" - they're sharing the same original)
Truncating will remove the `cloneOfCloneErrataA` from `channelC` as it's not in `channelB` nor is it's original `errataA`.
* Scenario 2: `cloneOfCloneErrataA` original points to `cloneErrataA`
Truncating will remove the `cloneOfCloneErrataA` from `channelC` as it's not in `channelB` nor is it's original `cloneErrataA`

In both cases the `channelC` ends up with no errata, which is what we want. IOW the method does, what the documentation is saying.

Another possible scenario would be this:
* the channel/errata configuration as you described
* the `cloneOfCloneErrataA` original points to `errataA`
* truncating `channelC` to `channelB`
In this case, the `cloneOfCloneA` would be removed from the `channelC`, which is maybe not what we would want, however, this is a documented behavior according to `truncateErrata` doc (the source channel (`channelB`) does not contain `cloneOfCloneErrataA` nor `cloneOfErrataA`).

Bottom line is:
* the Content Scenario will work fine with the current implementation,
* the "brothers" scenario (the last one) works as documented/expected.

File: java/code/src/com/redhat/rhn/manager/errata/ErrataManager.java [hunk 2]
```diff
@@ -245,8 +245,6 @@ public static Errata addChannelsToErrata(Errata errata,
         //Save the errata
         log.debug("addChannelsToErrata - storing errata");
         storeErrata(errata);
-
-        errata = (Errata) HibernateFactory.reload(errata);
```
[the remark this answers is not in the document]
Comment (hustodemon) [reply java/code/src/com/redhat/rhn/manager/errata/ErrataManager.java]: In the end I decided to rollback the patch. I couldn't reproduce the issue (multiple objects with same identifier in the hibernate session) anymore.

File: java/code/src/com/redhat/rhn/manager/errata/ErrataManager.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (ncounter) [reply java/code/src/com/redhat/rhn/manager/errata/ErrataManager.java]: Thanks a lot for the veeeery clear explanation. Now I am much more confident with this. :+1:

File: java/code/src/com/redhat/rhn/manager/errata/ErrataManager.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (ncounter) [reply java/code/src/com/redhat/rhn/manager/errata/ErrataManager.java]: It makes sense, yeah. Let's keep it in mind in case it happens again in future.

One last question: did you pushed the rollback already? I still see the changes in place at the moment

[the remark this answers is not in the document]
Comment (hustodemon) [reply java/code/src/com/redhat/rhn/manager/errata/ErrataManager.java]: good point - i forgot to push it. i'll just remove the patch and will force-push.
